### ISSUE TITLE
Export single csv file for multiple targeted peak detection lists

### ISSUE THREAD

**Issue by karlfraser** [mzmine/mzmine2#717]
### Basic information

* My operating system: Win 10
* My MZmine version: 2.33

### What happened

I am processing 30 files using a target list of ~560 mz/rt pairs.  It works really well. My problem is that when I want to export the final results of the 30 files peak lists as a single csv file, it will not write the results as a single file. I can select 'all peak lists' as an option but it still only exports the first one on the list as a csv file.  Also, when I try 'all selected in main window' - making sure I have them all selected, it still only writes a csv file with the first files results in it?

Please can someone tell me if this is possible to export multiple target list results as a single file.

Cheers, Karl

**Comment by Annexhc** [comment]
Hey Karl,
Do I understand you correctly, that you want all results of all lists in a single file? Than you should align your feature lists first and export only the aligned feature list.
The csv export cannot merge multiple feature lists to a single file. The function you are referring to enables the batch export of multiple feature lists to multiple csv files (check the tool tip of the filename parameter. It says something like use "{}" in file name...)
Also, I like to recommend you should update to the latest MZmine version.
best wishes,
Ansgar

**Comment by karlfraser** [comment]
Hi Ansgar,

Yes, you understand that correctly.  Thanks for your answer, I'm glad I'm not missing anything obvious there then.  I have tried the join aligner but what I find is even though this is a list of a known 560 targets, the aligner gives me a list of ~2600 peaks for what should actually be a list the same length as my target list. Is there a way to force it to use the ID only? I have found a tick box to force it to use the ID but it still gives me a list of 1400 peaks, so that doesnt seem to work? Is there anything obvious I am missing?

Thanks for your prompt reply too.

Cheers, Karl

**Comment by robinschmid** [comment]
Hey Karl.

What are your join aligner settings?
You might want to increase the m/z and rt tolerances.

Cheers
Robin

**Comment by karlfraser** [comment]
Hi Robin,

I am using:
m/z tol = 0.001 or 5ppm (it is data from a Q-tof with excellent mass stability)
Weight for m/z = 85
Rt tol = 0.3 min (absolute) - (the chromatogram is 12 mins long with good stability, normally around 0.1 min variance across the 30 runs)
Weight for Rt = 60
Requires same ID 

It is usually only a few peaks from each sample that dont align, but these differ across the batch and with a list of 564 targets and 30 samples, even if only even if only 1 in 50 peaks of my 560 target peaks dont match when you multiply by 30 samples that adds an extra 300+ falsely aligned peaks.

Cheers, Karl

**Comment by robinschmid** [comment]
Actually, I have never done any targeted analysis within MZmine. 
Do all of your 564 targets have an ID in the feature lists?
Then you could just set the tolerances to something really high and only depend on the ID to align.

Otherwise, what is your minimum m/z difference in your 564 targets? Maybe you can set the m/z tolerance higher, as your targets are far apart in a spectrum.

I tend to use a higher tolerance for q-TOF instruments but it always depends on your sample batch, method, and analysis.

Cheers

**Comment by karlfraser** [comment]
Yes they all have unique ID's in the mzmine feature lists - my target list is a targeted lipidomics list from supplementary info from a paper and I am trying to see how many I can detect with the Q-Tof.
I have tried many settings but cannot seem to force it to use only the ID. It seems that with the check box ticked to 'Require same ID', the other settings have an influence on merging the lists, which to me seems unusual. Can I suggest that for using mzmiine for targeted analysis, it would be really useful if it simply merged all the peak lists (of course each peak list has to have the same number of peaks and correctly matches the target list used) into a single matrix of sample vs target ID and contained either peak area or height as the variable reported in the output csv.

Cheers, Karl

**Comment by robinschmid** [comment]
Makes sense. The targeted approach should work like this. @tomas-pluskal  can you comment on this?

@karlfraser for lipids, did you also check out the non-target modules that are included in MZmine?
@Annexhc developed some lipid identification and visualization modules as there are plenty of isomers and lipids with similar m/z and rt (depending on your chromatography).
These modules directly check the MS/MS spectra for head groups and lipid chains.

Cheers

**Comment by karlfraser** [comment]
I have downloaded the latest version and read a couple of the posts on here around the lipid modules. Exciting stuff. We have collected SWATH type MS2 data on the Q-Tof (Data independent analysis), so I wonder if the module can handle that type of MS2 data, or does it need 'true' MS2 (i.e. a known parent ion fragmented to give a product ion spectrum as a MS2 spectrum)?
I have processed this data using MS-DIAL, and it seems to have identified a lot of lipids in the plasma sample - thus I also want to check in the MS1 spectrum my target list against what the MS-DIAL SWATH based identifications gave me.

Cheers, Karl

**Comment by Annexhc** [comment]
Hey Karl,
the lipid search module in MZmine is designed for DDA MS/MS data. I dont have tried SWATH data myself. Maybe we could extend the module for SWATH data in the future.

**Comment by robinschmid** [comment]
I don't know enough about your dataset so this is only a general remark.
SWATH has a very high potential, however, I do not know if it shows utility for lipidomics data. You will always have different co-eluting lipids, with different precursor m/z. In SWATH you fragment all of them together and you might find some false head group to chain compositions.
